sql/moz-fx-data-shared-prod/udf_js/bootstrap_percentile_ci/udf.sql

```diff
@@ -0,0 +1,337 @@
+        "metric": metric,
+        "statistic": "percentile",
+        "lower": lower,
+        "point": point,
+        "upper": upper,
+        "parameter": percentile,
+      });
+    }
+    return results;
+  }
+  return getPercentilesWithCI(percentiles, histogram, metric);
+  """;
+
+-- Tests
+WITH test_data AS (
+  SELECT
+    'test' AS metric_name,
+    STRUCT(
+      [
+        STRUCT(0.0 AS key, 0.0 AS value),
+        STRUCT(1.0 AS key, NULL AS value),
+        STRUCT(2.0 AS key, 0.0 AS value)
+      ] AS values
+    ) AS null_value,
+    STRUCT(
+      [
+        STRUCT(0.0 AS key, 0.0 AS value),
+        STRUCT(NULL AS key, 10.0 AS value),
+        STRUCT(2.0 AS key, 0.0 AS value)
+      ] AS values
+    ) AS null_key,
+    STRUCT(
+      [
+        STRUCT(0.0 AS key, 0.0 AS value),
+        STRUCT(1.0 AS key, 0.0 AS value),
+        STRUCT(2.0 AS key, 0.0 AS value)
+      ] AS values
+    ) AS all_zero,
+    STRUCT(
+      [
+        STRUCT(0.0 AS key, 0.0 AS value),
+        STRUCT(1.0 AS key, 10.0 AS value),
+        STRUCT(2.0 AS key, 0.0 AS value)
+      ] AS values
+    ) AS one_bin,
+    STRUCT(
+      [
+        STRUCT(-4.5 AS key, 0.0 AS value),
+        STRUCT(-3.5 AS key, 1.0 AS value),
+        STRUCT(-2.5 AS key, 12.0 AS value),
+        STRUCT(-1.5 AS key, 22.0 AS value),
+        STRUCT(-0.5 AS key, 30.0 AS value),
+        STRUCT(0.5 AS key, 22.0 AS value),
+        STRUCT(1.5 AS key, 12.0 AS value),
+        STRUCT(2.5 AS key, 1.0 AS value),
+        STRUCT(3.5 AS key, 0.0 AS value)
+      ] AS values
+    ) AS seven_bin
+),
+all_zero_percentiles AS (
+  SELECT
+    udf_js.bootstrap_percentile_ci([5, 25, 50, 75, 95], all_zero, metric_name) AS output
+  FROM
+    test_data
+),
+all_zero_results AS (
+  SELECT
+    output.*
+  FROM
+    all_zero_percentiles,
+    UNNEST(output) AS output
+),
+one_bin_percentiles AS (
+  SELECT
+    udf_js.bootstrap_percentile_ci([5, 25, 50, 75, 95], one_bin, metric_name) AS output
+  FROM
+    test_data
+),
+one_bin_results AS (
+  SELECT
+    output.*
+  FROM
+    one_bin_percentiles,
+    UNNEST(output) AS output
+),
+seven_bin_percentiles AS (
+  SELECT
+    udf_js.bootstrap_percentile_ci([5, 25, 50, 75, 95], seven_bin, metric_name) AS output
+  FROM
+    test_data
+),
+seven_bin_results AS (
+  SELECT
+    output.*
+  FROM
+    seven_bin_percentiles,
+    UNNEST(output) AS output
+)
+-- valid outputs
+SELECT
+  assert.equals('test', (SELECT DISTINCT metric FROM all_zero_results)),
+  assert.equals('percentile', (SELECT DISTINCT statistic FROM all_zero_results)),
+  assert.array_equals(
+    (SELECT ['5', '25', '50', '75', '95']),
+    (SELECT ARRAY(SELECT parameter FROM all_zero_results))
+  ),
+  assert.null((SELECT ARRAY(SELECT lower FROM all_zero_results WHERE lower IS NOT NULL))),
+  assert.null((SELECT ARRAY(SELECT point FROM all_zero_results WHERE point IS NOT NULL))),
+  assert.null((SELECT ARRAY(SELECT upper FROM all_zero_results WHERE upper IS NOT NULL))),
```

Comment:
I made an update to these tests since they were failing with ` Array cannot have a null element; error in writing field f0_ ` previously.